framework/src/mesh/FaceInfo.C

```diff
@@ -41,6 +41,32 @@ FaceInfo::FaceInfo(const ElemInfo * elem_info, unsigned int side, const dof_id_t
   _face_centroid = face->vertex_average();
 }
 
+FaceInfo::FaceInfo(const ElemInfo * elem_info,
```

Comment:
You can reduce the duplication here by defining a common private constructor that builds everything here given a FE object and a side, and then having your main two constructors call that constructor